choreographer/pipe.py

```diff
@@ -41,6 +42,9 @@ def __init__(self, debug=False, cls=NumpyEncoder):
         self.debug = debug
         self.cls=cls
 
+        # this is just a convenience to prevent multiple shutdowns
+        self.shutdown_lock = Lock()
```

Comment:
Why do you use other Lock for Pipe and doesn't use the Lock from Browser here?

Reply:
ohhh, it's for the theads

Reply:
`Pipe()` actually doesn't do anything with asyncio, and also we don't block on the lock, we use it as an atomic flag, so we can use actually any lock we want, I just like the threaing.lock() api more

`Browser()` has a switch between asyncio and sync

Reply:
Ohhh is for the Threads